### PR TITLE
Test cleanup

### PR DESCRIPTION
**Pull request by lucianoviana**

### CODE REVIEW

File: artio-system-tests/src/test/java/uk/co/real_logic/artio/system_tests/PersistentSequenceNumberGatewayToGatewaySystemTest.java [hunk 1]
```diff
@@ -1134,6 +1147,16 @@ private void exchangeMessagesAroundARestart(
             assertSequenceFromInitToAcceptAt(expectedInitToAccSeqNum, expectedAccToInitSeqNum);
         }
 
+        // this means a resend request will be sent by the acceptor
+        if (initialSentSequenceNumber > expectedAccToInitSeqNum)
```
Comment (wojciech-adaptive) [inline artio-system-tests/src/test/java/uk/co/real_logic/artio/system_tests/PersistentSequenceNumberGatewayToGatewaySystemTest.java:1151]: Aren't you comparing sequence numbers across two different directions?

**Review by wojciech-adaptive — CHANGES_REQUESTED** [review]

File: artio-binary-entrypoint-system-tests/src/test/java/uk/co/real_logic/artio/system_tests/BinaryEntryPointClient.java [hunk 2]
```diff
@@ -89,6 +90,8 @@ public final class BinaryEntryPointClient implements AutoCloseable
     private long keepAliveIntervalInMs = KEEP_ALIVE_INTERVAL_IN_MS;
     private CancelOnDisconnectType cancelOnDisconnectType = DO_NOT_CANCEL_ON_DISCONNECT_OR_TERMINATE;
     private long codTimeoutWindow = DeltaInMillisEncoder.timeNullValue();
+    private static final OtfHeaderDecoder OTF_HEADER_DECODER = new OtfHeaderDecoder(
+        BinaryEntryPointProtocol.loadSbeIr().headerStructure());
```
[the remark this answers is not in the document]
Comment (lucianoviana) [reply artio-binary-entrypoint-system-tests/src/test/java/uk/co/real_logic/artio/system_tests/BinaryEntryPointClient.java:94]: this is what the `JsonPrinter.print` method uses. I've just reproduced the issue and it happens when we run `shouldDisconnectConnectionWhenOutOfRangeTemplateIdUsed()` test with FIX_TEST log tag enabled.

File: artio-binary-entrypoint-system-tests/src/test/java/uk/co/real_logic/artio/system_tests/CancelOnDisconnectBinaryEntrypointSystemTest.java [hunk 3]
```diff
@@ -216,7 +216,7 @@ private void assertTriggersCancelOnDisconnect(final long logoutTimeInNs)
         assertEquals(onlySession.key(), result.context.key());
         final long timeoutTakenInNs = result.timeInNs - logoutTimeInNs;
         assertThat(timeoutTakenInNs, greaterThanOrEqualTo(codTimeoutInNs));
-        assertEquals(1, timeoutHandler.invokeCount());
+        testSystem.await("timeoutHandler.invokeCount() is not 1", () -> 1 == timeoutHandler.invokeCount());
```
[the remark this answers is not in the document]
Comment (lucianoviana) [reply artio-binary-entrypoint-system-tests/src/test/java/uk/co/real_logic/artio/system_tests/CancelOnDisconnectBinaryEntrypointSystemTest.java:219]: as I said privately I'd prefer refactoring this await method later on as it's being used in a bunch of tests - for now i'm just reusing it.

File: artio-system-tests/src/test/java/uk/co/real_logic/artio/system_tests/PersistentSequenceNumberGatewayToGatewaySystemTest.java (hunk 1 above)
Comment (lucianoviana) [reply artio-system-tests/src/test/java/uk/co/real_logic/artio/system_tests/PersistentSequenceNumberGatewayToGatewaySystemTest.java:1151]: yeh - am fixing the logic - thx

File: artio-binary-entrypoint-system-tests/src/test/java/uk/co/real_logic/artio/system_tests/BinaryEntryPointClient.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (lucianoviana) [reply artio-binary-entrypoint-system-tests/src/test/java/uk/co/real_logic/artio/system_tests/BinaryEntryPointClient.java:94]: this is what's used by JsonPrinter.print

**Pull request merged**